activerecord/lib/active_record/associations/belongs_to_association.rb

```diff
@@ -11,8 +11,15 @@ def handle_dependency
         when :destroy
           raise ActiveRecord::Rollback unless target.destroy
         when :destroy_async
-          id = owner.public_send(reflection.foreign_key.to_sym)
-          primary_key_column = reflection.active_record_primary_key.to_sym
+          primary_key_column = nil
+          id = nil
+          if reflection.foreign_key.is_a?(Array)
+            primary_key_column = Array(reflection.active_record_primary_key).map(&:to_sym)
+            id = Array(reflection.foreign_key).map { |col| owner.public_send(col.to_sym) }
```

Comment:
We already have an array so we don't need to wrap it. Although I think we can just do these two lines and remove the `is_a?` check as well as the `else` block.

Reply:
The `if reflection.foreign_key.is_a?(Array)` branch was my suggestion as I don't think we can pass array-based values into `enqueue_destroy_association` as an application may have a custom-defined job for deletion which is not ready to accept `association_primary_key_column` and `association_ids` as arrays (or as an array of arrays)  

I assumed that we don't want to introduce such a breaking change without a deprecation cycle. But this was the only concern so if we can allow breaking manually defined jobs then `reflection.foreign_key.is_a?(Array)` won't be necessary.

Reply:
As Nikita alluded to, the motivation was to maintain the existing signature for the destroy job. By branching, we'll only modify it in the query_constraints context, which is opt-in.

What your thoughts about that concern?

Reply:
I've maintained the branching, but strengthened the assumptions that are made in the branch. Assuming we're in a `query_constraints_list` context, we know the data structures of interest are arrays. Commit aef885c simplifies the code quite a bit, and makes branching less of a concern.

Reply:
Oh right, thanks for walking me through that. I totally forgot these were jobs. Correct we don't want a breaking change without deprecation, especially in jobs. 

Changes here look good. 👍🏼